File: Data Structure/Array/array.py
# Data Structure
# Array

class Array:
    
    # Constructor
    def __init__(self, size):
        self.size = size
        self.arr = [0] * size
        
    # Destructor
    def __del__(self):
        del self.arr
        
    # Get the size of the array
    def getSize(self):
        return self.size
        
    def insert(self, index, value):
        if index < 0 or index > self.size:
            return -1
        self.arr[index] = value
        return 0
        
    def remove(self, index):
        self.arr[index] = 0
        
    def reverse(self):
        for i in range(self.size // 2):
            temp =  self.arr[i]
            self.arr[i] =  self.arr[self.size - i - 1]
            self.arr[self.size - i - 1] = temp
            
    def print(self):
        print("Values inside array: ")
        for i in self.arr:
            print(i, end = " ")
        print()
        
# main 
if __name__ == "__main__":

    arr = Array(10)
    print("Array size: ", arr.getSize())
    
    arr.print()
    
    for index in range(len(arr.arr)):
        arr.insert(index, index)
        
    arr.print()
    
    arr.remove(5)
    
    arr.print()
    
    arr.reverse()
    
    arr.print()

File: Data Structure/Array/array.cpp
// Data Structure
// Array

#include <iostream>

class Array {
private:
    int *arr;
    int size;

public:
    // constructor
    Array (int size) {
        this->size = size; // set size into class variable
        arr = new int[size]; // dynamic array
        for (int i = 0; i < size; i++) {
            arr[i] = 0;
        }
    }
    
    // destructor
    ~Array() {
        delete[] arr;
    }

    int insert (int index, int value) {
        if (index < 0 || index > size) {
            return -1;
        }
        arr[index] = value;
        
        return 0;
    }
    
    int remove (int index) {
        if (index < 0 || index > size) {
            return -1;    
        }
        arr[index] = 0;
        
        return 0;
    }
    
    int print () {
        std::cout << "Values inside array: ";
        for (int i = 0; i < size; i++) {
            std::cout << arr[i] << " ";
        }
        std::cout << "\n";
        return 0;
    }
    
    int getSize() {
        return size;
    }
    
    void reverse() {
        int temp;
        for (int i = 0; i < size/2; i++) {
            temp = arr[i];
            arr[i] = arr[size - i - 1];
            arr[size - i - 1] = temp;
        }
    }
};

int main(int argc, char** argv) {
    
    // call class Array
    // initalize array with size 10
    Array arr(10);
    
    std::cout << "Array size: " << arr.getSize() << std::endl;
    
    arr.print(); // print array
    
    for (int index = 0; index < arr.getSize(); index++)
        arr.insert(index, index);
        
    arr.print(); // print array
    
    arr.remove(5); // remove value at index 5
    
    arr.print(); // print array
    
    arr.reverse(); // reverse array
    
    arr.print(); // print array
    
    return 0;
}